solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -500,6 +508,31 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
     }
   }
 
+  private void tagRequestWithRequestId(ResponseBuilder rb) {
+    final boolean ridTaggingDisabled = rb.req.getParams().getBool(CommonParams.DISABLE_REQUEST_ID, false);
+    if (! ridTaggingDisabled) {
+      String rid = getRequestId(rb.req);
+      if (StringUtils.isBlank(rb.req.getParams().get(CommonParams.REQUEST_ID))) {
+        ModifiableSolrParams params = new ModifiableSolrParams(rb.req.getParams());
```

Comment:
Maybe we don't need a flag parameter to decide what to do with rid.  What if Solr sees rid=\* (or true?) then it knows to generate a rid, and if it sees a blank/empty string (or false?) then it does nothing.  If it's something else than it *is* the rid (either from the client or generated).  One parameter to look at for this feature seems nicer than two.  Also, relating to my concern about wholesale disabling, it could be set in the solrconfig as a param default.

Reply:
I like the simplicity of having the flag be its own separate parameter over packing in multiple different "flag" values to the existing param.

Simplicity and intuitiveness are admittedly subjective though, I'm open to your opinion if you can convince me why you see having only a single param is easier for users to digest.

Reply:
Parameter / cognitive overload / excessive.

If a user passes `rid=whatever` then Solr will pass this along (thanks to you); right?  Then in a sense it's self-disableable by a user with a blank rid and putting it in solrconfig.xml init param for `/*`.

Reply:
I guess I just see that as being more complex.  Imagine describing that behavior to a newcomer in one sentence: "Well if you don't specify it at all we do X, if you do specify it we do Y, and if you specify it as blank we do Z".  Versus: "It either autogenerates a request ID or uses whatever you pass in.  There's also a flag to disable it if you want".

I'm the first to admit these sort of takes are subjective though.  If someone else chimes in here and I get outnumbered I'll change it for sure.  But I'll stick to my guns for now.

Reply:
> "It either autogenerates a request ID or uses whatever you pass in. There's also a flag to disable it if you want"

That's also an X, Y, and Z :-)

In my suggestion, the consequence of a blank rid needn't be a special case -- treat it as valid if you like.  It's super-short "rid=" and so adds negligible noise when propagated.